### PR TITLE
Fix for alma issue: show all files

### PR DESCRIPTION
**Pull request by keflavich**
Issue #1681 notes that we don't show all the files in the tarball, but we optionally can do that.

Now that the deployment was "succesful", this PR also cleans out the old cruft...
(I should have done this in two commits...)

### CODE REVIEW

**Comment by bsipocz** [comment]
@keflavich - edited the title, so I have some chance to know what module this is about when it pops up in my notifications 😊

**Comment by bsipocz** [comment]
also, ping me when it's ready for review.

**Comment by keflavich** [comment]
This is ready for review

File: astroquery/alma/core.py [hunk 1]
```diff
@@ -249,6 +249,13 @@ def stage_data(self, uids):
         uids : list or str
             A list of valid UIDs or a single UID.
             UIDs should have the form: 'uid://A002/X391d0b/X7b'
+        expand_tarfiles : bool
+            Expand the tarfiles to obtain lists of all contained files?  If
```
Comment (bsipocz) [inline astroquery/alma/core.py:253]: Why the '?'

File: astroquery/alma/core.py [hunk 2]
```diff
@@ -280,33 +287,50 @@ def stage_data(self, uids):
...
+                                        name='isProprietary'))
+
+                tables.append(table)
+                log.debug("Completed metadata retrieval for {0}".format(uu))
 
         if len(tables) == 0:
             raise ValueError("No valid UIDs supplied.")
 
+        if return_json:
+            return table
```
Comment (bsipocz) [inline astroquery/alma/core.py:332]: shouldn't this be `tables` ?

File: astroquery/alma/core.py [hunk 3]
```diff
@@ -280,33 +287,50 @@ def stage_data(self, uids):
...
+                                        name='URL'))
+
+                isp = self.is_proprietary(uid)
+                table.add_column(Column(data=[isp for row in table],
+                                        name='isProprietary'))
+
+                tables.append(table)
+                log.debug("Completed metadata retrieval for {0}".format(uu))
 
         if len(tables) == 0:
```
Comment (bsipocz) [inline astroquery/alma/core.py:328]: Indent looks bad here, but I'm certain it's just a bug in the github app.

File: astroquery/alma/core.py [hunk 4]
```diff
@@ -280,33 +287,50 @@ def stage_data(self, uids):
                     # this indicates a wrong server is being used;
                     # the "pre-feb2020" stager will be phased out
                     # when the new services are deployed
-                    return self.stage_data_prefeb2020(uids)
+                    raise ValueError("Failed query!  This shouldn't happen - please "
```
Comment (bsipocz) [inline astroquery/alma/core.py:290]: Should this be ServerError, or one of the more specific errors we have,  instead plain ValueError? (Given the user is unlikely can do anything about it)

File: astroquery/alma/core.py (hunk 1 above)
Comment (keflavich) [reply astroquery/alma/core.py:253]: Well, it's a boolean, so I posed it as a question, apparently.
```suggestion
            Expand the tarfiles to obtain lists of all contained files.  If
```

File: astroquery/alma/core.py (hunk 4 above)
Comment (keflavich) [reply astroquery/alma/core.py:290]: yes

File: astroquery/alma/core.py (hunk 3 above)
Comment (keflavich) [reply astroquery/alma/core.py:328]: yeah, I don't see a problem

File: astroquery/alma/core.py (hunk 2 above)
Comment (keflavich) [reply astroquery/alma/core.py]: yes, good catch, that would have been a pain to find

**Comment by bsipocz** [comment]
Could you add a tests and changelog, then this should be good to go.

**Comment by bsipocz** [comment]
You really need to enable a flake8 plugin :trollface: 

Otherwise, one more test for `return_json`, and this is good to go.

**Comment by keflavich** [comment]
the json return doesn't really need a test, it's there primarily for debug access.... then again... it was broken... ok fine.

**Comment by bsipocz** [comment]
Ok, this horse is beaten for long enough.

Thanks @keflavich for the cleanup!

**Pull request merged**